pkg/sqlite/query.go

```diff
@@ -541,3 +540,104 @@ func (s *SQLQuerier) GetApisForEntry(ctx context.Context, entryId int64) (provid
 
 	return
 }
+
+func (s *SQLQuerier) GetBundleVersion(ctx context.Context, image string) (string, error) {
+	query := `SELECT version FROM operatorbundle WHERE bundlepath=? LIMIT 1`
+	rows, err := s.db.QueryContext(ctx, query, image)
+	if err != nil {
+		return "", err
+	}
+	defer rows.Close()
+
+	var version sql.NullString
+	if rows.Next() {
+		if err := rows.Scan(&version); err != nil {
+			return "", err
+		}
+	}
+	if version.Valid {
+		return version.String, nil
+	}
+	return "", fmt.Errorf("bundle %s not found", image)
+}
+
+func (s *SQLQuerier) GetImagesForPackage(ctx context.Context, pkgName string) ([]string, error) {
+	query := `SELECT DISTINCT image FROM operatorbundle 
+	INNER JOIN channel ON operatorbundle.name=channel.head_operatorbundle_name
+	WHERE channel.package_name=?`
+	rows, err := s.db.QueryContext(ctx, query, pkgName)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	images := []string{}
+	for rows.Next() {
+		var imgName sql.NullString
+		if err := rows.Scan(&imgName); err != nil {
+			return nil, err
+		}
+		if imgName.Valid {
+			images = append(images, imgName.String)
+		}
+	}
+	return images, nil
+}
+
+func (s *SQLQuerier) GetDefaultChannelForPackage(ctx context.Context, pkgName string) (string, error) {
+	query := `SELECT DISTINCT default_channel FROM package WHERE name=?`
+	rows, err := s.db.QueryContext(ctx, query, pkgName)
+	if err != nil {
+		return "", err
+	}
+	defer rows.Close()
+
+	var defaultChannel sql.NullString
+	if rows.Next() {
+		if err := rows.Scan(&defaultChannel); err != nil {
+			return "", err
+		}
+	}
+	if defaultChannel.Valid {
+		return defaultChannel.String, nil
+	}
+	return "", nil
+}
+
+func (s *SQLQuerier) ListChannels(ctx context.Context, pkgName string) ([]string, error) {
+	query := `SELECT DISTINCT name FROM channel WHERE channel.package_name=?`
+	rows, err := s.db.QueryContext(ctx, query, pkgName)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	channels := []string{}
+	for rows.Next() {
+		var chName sql.NullString
+		if err := rows.Scan(&chName); err != nil {
+			return nil, err
+		}
+		if chName.Valid {
+			channels = append(channels, chName.String)
+		}
+	}
+	return channels, nil
+}
+
+func (s *SQLQuerier) GetCurrentCSVForChannel(ctx context.Context, pkgName, channel string) (string, error) {
```

Comment:
This just gets the name. The existing `GetBundle` will give you the full bundle contents (if it exists)

Reply:
I only need the name to generate the package.yaml